### PR TITLE
fix: remove closed listeners from transport manager

### PR DESCRIPTION
**Pull request by achingbrain**
When a listener closes, remove it from the stored list of listeners.

This wasn't much of a problem before as listeners tended to be very long lived but circuit relay listeners get closed all the time so this fixes a now much more obvious memory leak.

### CODE REVIEW

File: src/transport-manager.ts [hunk 1]
```diff
@@ -216,6 +223,11 @@ export class DefaultTransportManager implements TransportManager, Startable {
             })
         })
         listener.addEventListener('close', () => {
```
Comment (maschad) [inline src/transport-manager.ts:225]: should we be removing these eventListeners once they are closed as well?

**Review by maschad — COMMENTED** [review]
Left a comment, LGTM otherwise

File: src/transport-manager.ts (hunk 1 above)
Comment (achingbrain) [reply src/transport-manager.ts:225]: I don't think so because the listener is created and returned by the transport but not retained.

**Review by maschad — APPROVED** [review]
LGTM

**Pull request merged**